test/core/xds/xds_client_ads_stream_wait_test.cc

```diff
@@ -59,21 +58,37 @@ TEST_F(XdsClientNotifyWatchersDone, Basic) {
           .set_nonce("A")
           .AddFooResource(XdsFooResource("foo1", 6))
           .Serialize());
+  stream->SendMessageToClient(
+      ResponseBuilder(XdsFooResourceType::Get()->type_url())
+          .set_version_info("2")
+          .set_nonce("A")
+          .AddFooResource(XdsFooResource("foo1", 8))
+          .Serialize());
   // XdsClient should have delivered the response to the watcher.
   auto resource = watcher->WaitForNextResourceAndHandle();
   ASSERT_NE(resource, absl::nullopt);
-  EXPECT_EQ(resource->first->name, "foo1");
-  EXPECT_EQ(resource->first->value, 6);
+  EXPECT_EQ(resource->name(), "foo1");
+  EXPECT_EQ(resource->resource_value(), 6);
   // XdsClient should have sent an ACK message to the xDS server.
   request = WaitForRequest(stream.get());
   ASSERT_TRUE(request.has_value());
   CheckRequest(*request, XdsFooResourceType::Get()->type_url(),
                /*version_info=*/"1", /*response_nonce=*/"A",
                /*error_detail=*/absl::OkStatus(),
                /*resource_names=*/{"foo1"});
-  EXPECT_EQ(stream->read_count(), 0);
-  resource->second.reset();
   EXPECT_EQ(stream->read_count(), 1);
+  resource = absl::nullopt;
+  EXPECT_EQ(stream->read_count(), 2);
+  resource = watcher->WaitForNextResourceAndHandle();
+  ASSERT_NE(resource, absl::nullopt);
```

Comment:
After this, please verify that we actually get the expected value for the resource.

Reply:
Done